db/migrate/20180310000000_add_last_read_notification_id_to_users.rb

```diff
@@ -0,0 +1,25 @@
+require Rails.root.join('lib', 'mastodon', 'migration_helpers')
+
+class AddLastReadNotificationIdToUsers < ActiveRecord::Migration[5.1]
+  include Mastodon::MigrationHelpers
+
+  disable_ddl_transaction!
+
+  def change
+    add_column :users, :last_read_notification_id, :bigint
+
+    reversible do |dir|
+      dir.up do
+        safety_assured do
+          add_column_with_default :oauth_access_tokens, :reading_notifications, :boolean, allow_null: false, default: false
+          update_column_in_batches :users, :last_read_notification_id, 9e9
```

Comment:
Would it not be easier to use -1 as the "magic value"? I am actually not sure if the space requirements for these numbers are equal or not, but I have a distrust of "max integer"